addons/hr_timesheet/migrations/10.0.1.0/post-migration.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def migrate_allow_timesheets(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE project_project pp
+        SET allow_timesheets = True
+        FROM account_analytic_account aaa
+        WHERE aaa.id = pp.analytic_account_id
+        AND aaa.%s
+        """ % openupgrade.get_legacy_name('use_timesheets'),
+    )
+
+
+def fill_analytic_line_project(env):
+    """Fill project with the linked one in the related task or issue."""
```

Comment:
No in the case of task_id, as hr_timesheet depends on project and it's the one that adds these fields in the account.analytic.line (see https://github.com/odoo/odoo/blob/c61ae3a3ed65b7a40b470076c5dbbd881e791f78/addons/hr_timesheet/hr_timesheet.py#L10). For field issue_id, I have to check.